ibis/common/grounds.py

```diff
@@ -38,6 +38,17 @@ def __create__(cls, *args, **kwargs):
         return type.__call__(cls, *args, **kwargs)
 
 
+class Immutable(Hashable):
```

Comment:
What is the default implementation of `__hash__` here? Curious how `Expr.__hash__` works after you removed its override.

Reply:
There is no default implementation, the subclasses must define `__hash__`.

`Expr` implements it:

```py
    def __hash__(self):
        return hash((self.__class__, self._arg))
```